scipy/stats/tests/test_stats.py

```diff
@@ -3711,8 +3711,8 @@ def test_1samp_ci_iv(self, xp):
     def test_pvalue_ci(self, alpha, data_axis, alternative, xp):
         # test relationship between one-sided p-values and confidence intervals
         data, axis = data_axis
-        data = data.astype(copy=True)  # ensure byte order
-        data = xp.asarray(data)
+        data = data.astype(np.float64, copy=True)  # ensure byte order
+        data = xp.asarray(data, dtype=xp.float64)
```

Comment:
Am I missing something with the "ensure byte order" comment? Can we not just include the `copy` in `asarray`?

```suggestion
        data = xp.asarray(data, dtype=xp.float64, copy=True)
```

Reply:
`hypothesis` generates them in some nonstandard format that `torch.asarray` won't accept

Reply:
okay, if that isn't possible then I think this looks okay. Maybe worth adding a comment as to why.

Reply:
I adjusted the test strategy to generate normal `float64`s. I think that the old strategy must have been left over from before the test took its final form.